src/option_parser.cr

```diff
@@ -59,11 +59,18 @@ class OptionParser
   protected property flags : Array(String)
   protected property handlers : Array(Handler)
   protected property unknown_args
+  protected property! missing_option : String ->
+  protected property! invalid_option : String ->
 
   # Creates a new parser.
   def initialize
     @flags = [] of String
     @handlers = [] of Handler
+    missing_option { |flag| raise MissingOption.new(flag) }
+    invalid_option { |flag| raise InvalidOption.new(flag) }
+    # TODO: these should work and remove the need for the property! calls above
+    # @missing_option = ->(option : String) { raise MissingOption.new(option) }
+    # @invalid_option = ->(option : String) { raise InvalidOption.new(option) }
```

Comment:
Fixed! Of course we can't immediately use this, but please check if the new compiler now compiles this fine :-)